ouroboros-consensus/docs/report/chapters/future/genesis.tex

```diff
@@ -1499,3 +1500,178 @@ \subsection{Possible optimisations}
+\end{itemize}
+\end{definition}
+
+There is one subtlety here: we are checking the $k$ limit when we instruct
+blockfetch to download blocks, but this does not automatically mean that when
+the chain database is presented with those blocks by blockfetch, the property
+still holds. Fortunately, we can argue that it does.
+
+Let $S$ be the suboptimal chain presented to block fetch and $O$ the optimal
+chain at the time chainsync made its decision. We need to consider whether by
+the time the chain database runs chain selection, there is a \emph{new} optimal
+chain $O'$ such that adopting $S$ would make it impossible to adopt $O'$, where
+it \emph{would} have been possible to switch from $O$ to $O'$. If the
+intersection between $O$ and $O'$ is \emph{after} the intersection between $O$
+and $S$, this is trivial:
+
+\begin{center}
+\begin{tikzpicture}[yscale=0.5]
+\draw (0,0) node{$\bullet$} -- ++(1,1) -- ++(3,0) node{$\bullet$} node[right]{$O$};
+\draw (0,0) -- ++(1,-1) -- ++(2,0) node{$\bullet$} node[right]{$S$};
+\draw (1,1) node{$\bullet$} -- ++(1,1) -- ++(3,0) node{$\bullet$} node[right]{$O'$};
+\end{tikzpicture}
+\end{center}
+
+Since we can roll back from $S$ to the intersection point with $O$, we can
+clearly also switch from $S$ to $O'$. If the intersection is further back,
+the situation is a bit more subtle:
+
+\begin{center}
+\begin{tikzpicture}[yscale=0.5]
+\draw (0,0) node{$\bullet$} -- ++(1,1) -- ++(3,0) node{$\bullet$} node[right]{$O$};
+\draw (0,0) -- ++(1,-1) -- ++(2,0) node{$\bullet$} node[right]{$S$};
+\draw (-1,1) node{$\bullet$} -- (0,0);
+\draw (-1,1) -- ++(1,1) -- ++(5,0) node{$\bullet$} node[right]{$O'$};
+\end{tikzpicture}
+\end{center}
+
+By assumption we can switch from $O$ to $O'$. This means that the distance from
+$O$ to the intersection with $O'$ is $k$ blocks or less. Moreover, $O$ must be
+at least as long as $S$ (otherwise $O$ could not be preferred over $S$, and $S$
+would not be suboptimal). This means that the distance from $S$ to the
+intersection with $O'$ must also be at most $k$, and hence we can switch from
+$S$ to $O'$. This means that adoping $S$ does not prevent us from switching to
+any other chain $O'$ that we could have switched to from $O$. \qed
+
+Under exceptional circumstances it is possible that if we \emph{ourselves} forge
+a block on top of a suboptimal chain, the distance from that suboptimal chain to
+the intersecion with the optimal one might now exceed $k$. However, the
+``network respects $\Delta$'' principle saves us from having to treat this as a
+special case.
+
+\subsection{When the node is behind}
+
+To a certain extend, no special provisions need to be made when the node is
+behind. The rule that prohibits adopting more than $k$ blocks from suboptimal
+chains still guarantees that we can continue to switch to the optimal chain
+once we get its blocks; the only difference is that we might hit that limit
+much more quickly.
```

Comment:
>  the only difference is that we might hit that limit much more quickly.

I did not get this part.

Reply:
When we are up to date, we expect one block every 20 seconds. When we are bulk syncing, we expect many of blocks per second.